frontend/src/app/main/ui/workspace/right_header.cljs

```diff
@@ -169,33 +169,32 @@
 
         active-comments
         (mf/use-fn
+         (mf/deps layout)
          (fn []
            (st/emit! :interrupt
-                     (dw/clear-edition-mode))
-           ;; Delay so anything that launched :interrupt can finish
-           (ts/schedule 100 #(st/emit! (dw/select-for-drawing :comments)))))
+                     (dw/clear-edition-mode)
+                     (-> (dw/remove-layout-flag :document-history)
+                         (vary-meta assoc ::ev/origin "workspace-header"))
+                     (dw/select-for-drawing :comments))))
 
         toggle-comments
         (mf/use-fn
-         (mf/deps selected-drawtool)
+         (mf/deps selected-drawtool comments-local)
          (fn [_]
-           (when (contains? layout :document-history)
-             (st/emit! (-> (dw/remove-layout-flag :document-history)
-                           (vary-meta assoc ::ev/origin "workspace-header"))))
-
            (if (= :comments selected-drawtool)
-             (st/emit! :interrupt)
+             (do
+               (st/emit! :interrupt)
+               (when (some? comments-local)
+                 (st/emit! :interrupt)))
```

Comment:
Do you need emit interrupt two times?

Reply:
yes, one to stop the path edition other to stop the drawing mode

Reply:
This looks like the problem is in other side. There should never be a need for two interrupts, a single interrupt should interrupt everything.